spec/index.html

```diff
@@ -5462,29 +5462,38 @@ <h5>RDFterm-equal</h5>
-              arguments are both literal but are not the same RDF term <sup><a href=
-                                                                               "#func-RDFterm-equal-foot1" class="footnote">*</a></sup>; returns FALSE otherwise.
-              <code>term1</code> and <code>term2</code> are the same if any of the following is
+            <p>Returns TRUE if <code>term1</code> and <code>term2</code> are <em>equal</em> RDF terms as defined below;
+              produces a type error if the arguments are both literals with the same datatype that is unsupported by the implementation but are not the same RDF term <sup><a href="#func-RDFterm-equal-foot1" class="footnote">*</a></sup>;
+              returns FALSE otherwise.</p>
+            <p>
+              <code>term1</code> and <code>term2</code> are <em>equal</em> RDF terms if any of the following is
               true:</p>
             <ul>
               <li>
-                <span class="name">term1</span> and <span class="name">term2</span> are equivalent
-                <span class="IRI type">IRIs</span> as defined in <a data-cite=
-                                                                    "RDF12-CONCEPTS#section-Graph-URIref">6.4 RDF URI References</a> of
+                <span class="name">term1</span> is an <span class="IRI type">IRI</span> and <span class="name">term2</span> is an <span class="IRI type">IRI</span>
+                such that these two <span class="IRI type">IRIs</span> are equal as per the notion of <a data-cite=
+                                                                    "RDF12-CONCEPTS#dfn-iri-equality">IRI equality</a> of
                 [[RDF12-CONCEPTS]].
               </li>
               <li>
-                <span class="name">term1</span> and <span class="name">term2</span> are equivalent
-                <span class="literal type">literals</span> as defined in <a data-cite=
-                                                                            "RDF12-CONCEPTS#section-Literal-Equality">6.5.1 Literal Equality</a> of
+                <span class="name">term1</span> is a <span class="literal type">literal</span> and
+                <span class="name">term2</span> is a <span class="literal type">literal</span> such that
+                these two <span class="literal type">literals</span> are equal as per the notion of <a data-cite=
+                                                                            "RDF12-CONCEPTS#dfn-term-equal">Literal term equality</a> of
                 [[RDF12-CONCEPTS]].
               </li>
+              <li>
+                <span class="name">term1</span> is a <span class="literal type">literal</span> and
+                <span class="name">term2</span> is a <span class="literal type">literal</span>
+                such that the <a data-cite="RDF12-CONCEPTS#dfn-datatype-iri">datatype IRI</a>
+                of these two literals is the same, this IRI is in the set of
```

Comment:
```suggestion
                of these two literals is the same, both datatype IRIs are in the set of the
```

The datatypes can be different (example `xsd:integer` and `xsd:decimal`, or `xsd:integer` and `xsd:double`).

Reply:
@afs Isn't the case of different datatypes relevant only for numeric datatypes? If so, then I think we don't need to cover this case here because equivalence testing (i.e., the `=` operator) for numerics is not defined in terms of the `RDFterm-equal` function. Instead, the [operator mapping table in Sec.17.3](https://w3c.github.io/sparql-query/spec/#OperatorMapping) defines `=` for numerics using the [XPath function `op:numeric-equal`](https://www.w3.org/TR/xpath-functions/#func-numeric-equal).

Reply:
This text has to be general, including for additional datatypes. 

In extended types, xsd:duration has derived datatypes. Somewhere, there could be "astromonical time" mapping to the timeline, or datatypes definitions mapping to the RGB color space.

This defines the extensions of `=`, including knowing when value spaces are distinct and `=` is false.

Reply:
I was assuming that such special cases (i.e., cases with different datatypes but equivalent literal values) can be captured formally as SPARQL language extensions that are defined by adding further rows to the operator mapping table in Sec.17.3 rather than relying on the definition of the `RDFterm-equal` function. At least, that's my interpretation of the first paragraph in [Sec. 17.3.1 Operator Extensibility](https://w3c.github.io/sparql-query/spec/#operatorExtensibility).

However, if such cases were/are meant to be captured through `RDFterm-equal` as you describe it, then I think the sentence that we are discussing here needs to be improved to make this aspect more explicit. Currently, the sentence (with your suggested change already applied) says:

> term1 is a literal and term2 is a literal such that the [datatype IRI](https://w3c.github.io/rdf-concepts/spec/#dfn-datatype-iri) of these two literals is the same, both datatype IRIs are in the set of the [recognized datatype IRIs](https://w3c.github.io/rdf-concepts/spec/#dfn-recognized-datatype-iri), and the [literal values](https://w3c.github.io/rdf-concepts/spec/#dfn-literal-value) associated with both literals is the same.

To capture your interpretation we need two changes in this sentence:
1. The first condition ("such that the [datatype IRI](https://w3c.github.io/rdf-concepts/spec/#dfn-datatype-iri) of these two literals is the same") has to be removed. (Do you agree?)
2. The last condition ("the [literal values](https://w3c.github.io/rdf-concepts/spec/#dfn-literal-value) associated with both literals is the same") needs to be changed because---if I understand your interpretation right (let me know if not)---it is actually okay if the two literal values are not the same value, as long as they are considered as equivalent values.

Is this what you are suggesting?

Reply:
@afs please take a look at my previous comment above once you have time.

Reply:
It might be better to rename the function. `RDFterm-equal` puts too much on the "RDFterm" and so reads like 'sameTerm' which it isn't.

It should be "operator(RDFterm,RDFTerm)-equality-by-value" which is ungainly :-)

"equals" is used for same-value (equal in datatype range), to contrast with `sameTerm` (equal in datatype domain).

Suggestions for alternative names:
* `RDF-value-equals`
* `sameValue` (c.f. `sameTerm`)
* `value-equals`

Yes, the first condition has to be removed.

I don't follow your point 2. How is "same value" different from "equivalent values"? Two values are the same if and only if they are identical - same elements of the set that is the common range of the two datatype mappings.

The original "associated with both literals" was meant to mean the values of each datatype mapping.

`sameValue` for literals is whether `equality(  datatype1(lexical1) , datatype2(lexcial2) )` where `equality` is a partial function of identity in the common range of the datatype mappings - the intersection of the two value spaces of datatype1 and datatype2.

Value spaces may overlap - intersection.

If the values for the literals are not both in the common range, then return "false" without knowing anything more. A common case is where the values spaces are disjoint.

`sameValue` is different to other extensions in that it gets the heart of what a literal with datatype is in RDF. Extensions need to follow that.


Can we get some adequate text now, get the working draft updated, and then loop round to discussing changes to the updated text, including a longer discussion at the end of 17.4.1.7 (we have multiple audiences - users and implementers ... as well as spec-ites)?

Reply:
@afs Regarding your proposal to rename `RDFterm-equal`, I agree. I have created a separate issue for it: #25

Reply:
> If this is okay with you

Yes - that's good for merge. 

We (and I offer to do it) can revisit when we have a new baseline text to work from alongside #25.